test/e2e/node/kubelet.go

```diff
@@ -309,52 +301,48 @@ var _ = SIGDescribe("kubelet", func() {
-				// only 30 seconds here because framework.RunRC already waited for all pods to
-				// transition to the running status.
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
-				framework.ExpectNoError(err)
-				if resourceMonitor != nil {
-					resourceMonitor.LogLatest()
-				}
-
-				ginkgo.By("Deleting the RC")
-				framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
-				// Check that the pods really are gone by querying /runningpods on the
-				// node. The /runningpods handler checks the container runtime (or its
-				// cache) and  returns a list of running pods. Some possible causes of
-				// failures are:
-				//   - kubelet deadlock
-				//   - a bug in graceful termination (if it is enabled)
-				//   - docker slow to delete pods (or resource problems causing slowness)
-				start := time.Now()
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, 0, itArg.timeout)
-				framework.ExpectNoError(err)
-				e2elog.Logf("Deleting %d pods on %d nodes completed in %v after the RC was deleted", totalPods, len(nodeNames),
-					time.Since(start))
-				if resourceMonitor != nil {
-					resourceMonitor.LogCPUSummary()
-				}
+		ginkgo.It("should let Kubelet delete 10 pods per node with in 1 minute", func() {
+			totalPods := 10 * numNodes
+			ginkgo.By(fmt.Sprintf("Creating a RC of %d pods and wait until all pods of this RC are running", totalPods))
+			rcName := fmt.Sprintf("cleanup%d-%s", totalPods, string(uuid.NewUUID()))
+
+			err := framework.RunRC(testutils.RCConfig{
+				Client:       f.ClientSet,
+				Name:         rcName,
+				Namespace:    f.Namespace.Name,
+				Image:        imageutils.GetPauseImageName(),
+				Replicas:     totalPods,
+				NodeSelector: nodeLabels,
 			})
-		}
+			framework.ExpectNoError(err)
+			// Perform a sanity check so that we know all desired pods are
+			// running on the nodes according to kubelet. The timeout is set to
+			// only 30 seconds here because framework.RunRC already waited for all pods to
+			// transition to the running status.
+			err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
+			framework.ExpectNoError(err)
+			if resourceMonitor != nil {
+				resourceMonitor.LogLatest()
+			}
+
+			ginkgo.By("Deleting the RC")
+			framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
+			// Check that the pods really are gone by querying /runningpods on the
+			// node. The /runningpods handler checks the container runtime (or its
+			// cache) and  returns a list of running pods. Some possible causes of
+			// failures are:
+			//   - kubelet deadlock
+			//   - a bug in graceful termination (if it is enabled)
+			//   - docker slow to delete pods (or resource problems causing slowness)
+			start := time.Now()
+			err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, 0, 1*time.Minute)
```

Comment:
This check expects the nodes under test to have the kubelet api accessible on a default port.  Nothing in our docs say kubelet API access is out of bounds for conformance, let alone assuming the default port, but I'm wondering if this was an oversight on our part.

https://github.com/kubernetes/kubernetes/blob/b409073e99695ea35642a8194b9285ac12fd0cf8/test/e2e/node/kubelet.go#L56-L57

https://github.com/kubernetes/kubernetes/blob/b409073e99695ea35642a8194b9285ac12fd0cf8/test/e2e/framework/kubelet/kubelet_pods.go#L41

The previous `DeleteRCAndWaitForGC` call verifies all pods are gone from the control plane's perspective, which seems fine.

Reply:
Removed the check in 67f8c2c